### PR TITLE
feat: add C implementation for `math/base/special/binomcoefln`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds `C` implementation for [`math/base/special/binomcoefln`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/binomcoefln).

```c
double stdlib_base_binomcoefln( const int64_t n, const int64_t k );
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/binomcoefln/README.md [hunk 1]
```diff
@@ -83,7 +83,7 @@ var binomcoefln = require( '@stdlib/math/base/special/binomcoefln' );
 
 #### binomcoefln( n, k )
 
-Evaluates the [binomial coefficient][binomial-coefficient] of two integers `n` and `k`.
+Evaluates the natural logarithm of the [binomial coefficient][binomial-coefficient] of two integers `n` and `k`.
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/binomcoefln/README.md:86]: This comment was incorrect, I've changed it.

File: lib/node_modules/@stdlib/math/base/special/binomcoefln/test/test.js [hunk 2]
```diff
@@ -75,7 +76,7 @@ tape( 'the function evaluates the binomial coefficient for integers `m` and `k`'
 			t.equal( v, expected[ i ], 'returns '+expected[i]+' when provided m: '+arg1[i]+' and k: '+arg2[i] );
 		} else {
 			delta = abs( v - expected[ i ] );
-			tol = 5.0e-15 * abs( expected[ i ] );
+			tol = 21.5 * EPS * abs( expected[ i ] );
```
Comment (gunjjoshi) [inline lib/node_modules/@stdlib/math/base/special/binomcoefln/test/test.js:79]: I've changed this to use `EPS` here and in `test.native.js` as well.

File: lib/node_modules/@stdlib/math/base/special/binomcoefln/src/main.c [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		return stdlib_base_binomcoefln( -n + k - 1, k );
+	}
+	if ( k < 0 ) {
+		return STDLIB_CONSTANT_FLOAT64_NINF;
+	}
+	if ( k == 0 ) {
+		return 0.0;
+	}
+	if ( k == 1 ) {
+		return stdlib_base_ln( stdlib_base_abs( n ) );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/binomcoefln/src/main.c]:
```suggestion
		return stdlib_base_ln( stdlib_base_abs( (double)n ) );
```

File: lib/node_modules/@stdlib/math/base/special/binomcoefln/src/main.c [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	}
+	if ( n < k ) {
+		return STDLIB_CONSTANT_FLOAT64_NINF;
+	}
+	if ( n - k < 2 ) {
+		return stdlib_base_binomcoefln( n, n - k );
+	}
+
+	// Case: n - k >= 2
+	return -stdlib_base_ln( n + 1 ) - stdlib_base_betaln( n - k + 1, k + 1 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/binomcoefln/src/main.c]:
```suggestion
	return -stdlib_base_ln( (double)(n + 1) ) - stdlib_base_betaln( (double)(n - k + 1), (double)(k + 1) );
```

Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/binomcoefln/src/main.c]: Okay, so whenever we cast, we don't include spaces around parenthesis for the term to be casted as well. Thanks for catching!

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/binomcoefln/src/main.c]: @gunjjoshi That was my preference here, don't think this is a convention we have firmly established.

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/binomcoefln/src/main.c]: No, this is the convention. `(double)`, not `( double )`.

Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/binomcoefln/src/main.c]: Yes, this is fine. What I was confused about was between `(double)(n + 1)` and `(double)( n + 1 )`.

**Pull request merged**